compiler/noirc_evaluator/src/acir/tests/instructions/binary.rs

```diff
@@ -0,0 +1,631 @@
+    BLACKBOX::RANGE input: w4, bits: 8
+    ASSERT w4 = w3
+    ");
+}
+
+#[test]
+fn checked_mul_u8_with_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8, v2: u1):
+        enable_side_effects v2
+        v3 = mul v0, v1
+        return v3
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // w0 is multiplied by w1, then the reslt (w4) is multiplied by the predicate (w2)
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1, w2]
+    public parameters: []
+    return values: [w3]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    BLACKBOX::RANGE input: w2, bits: 1
+    ASSERT w4 = w0*w1
+    ASSERT w5 = w2*w4
+    BLACKBOX::RANGE input: w5, bits: 8
+    ASSERT w5 = w3
+    ");
+}
+
+#[test]
+fn div_u8_no_predicate_by_var() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8):
+        v2 = div v0, v1
+        return v2
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // - w3 is 1/w1
+    // - we check `0 = w1*w3 - 1` to ensure that an inverse must exist (so that w1 != 0)
+    // - w4 is w0/w1
+    // - w5 is w0 mod w1
+    // - with `w6 = w1 - w5 - 1` and the next range check we ensure that w5 < w1
+    // - `w5 = -w1*w4 + w0` can be read as `w0 = w1*w4 + w5`
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: [w2]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    BRILLIG CALL func: 0, inputs: [w1], outputs: [w3]
+    ASSERT 0 = w1*w3 - 1
+    BRILLIG CALL func: 1, inputs: [w0, w1], outputs: [w4, w5]
+    BLACKBOX::RANGE input: w4, bits: 8
+    BLACKBOX::RANGE input: w5, bits: 8
+    ASSERT w6 = w1 - w5 - 1
+    BLACKBOX::RANGE input: w6, bits: 8
+    ASSERT w5 = -w1*w4 + w0
+    ASSERT w4 = w2
+
+    unconstrained func 0: directive_invert
+    0: @21 = const u32 1
+    1: @20 = const u32 0
+    2: @0 = calldata copy [@20; @21]
+    3: @2 = const field 0
+    4: @3 = field eq @0, @2
+    5: jump if @3 to 8
+    6: @1 = const field 1
+    7: @0 = field field_div @1, @0
+    8: stop &[@20; @21]
+    unconstrained func 1: directive_integer_quotient
+    0: @10 = const u32 2
+    1: @11 = const u32 0
+    2: @0 = calldata copy [@11; @10]
+    3: @2 = field int_div @0, @1
+    4: @1 = field mul @2, @1
+    5: @1 = field sub @0, @1
+    6: @0 = @2
+    7: stop &[@11; @10]
+    ");
+}
+
+#[test]
+fn div_u8_no_predicate_by_constant() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8):
+        v1 = div v0, u8 7
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // - Note how w2, w3 and w4 are range-checked with less bits than 8
+    // - with `w4 = w3 + 1` and the next range check we ensure that 0 <= w3 < 8
+    // - `w3 = w0 - 7*w2` can be read as `w0 = 7*w2 + w3` which is the definition of divmod
```

Comment:
Could we add something how `w2` is the quotient result and `w3` is the modulo result from our quotient directive. Let's add divmod to cspell or change to something like "integer division"